src/main/java/mate/academy/bookapp/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,37 @@
+package mate.academy.bookapp.repository.book;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookapp.dto.BookSearchParametersDto;
+import mate.academy.bookapp.model.Book;
+import mate.academy.bookapp.repository.SpecificationBuilder;
+import mate.academy.bookapp.repository.SpecificationProviderManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private static final String ISBN_KEY = "isbn";
+    private static final String TITLE_KEY = "title";
+    private static final String AUTHOR_KEY = "author";
```

Comment:
better make these constants public and reuse them in Providers. You will have one place for changes in the future